src/execute_batch_sc.rs

```diff
@@ -1,133 +1,228 @@
+                }
+            };
+            file.read_to_end(&mut bytecode)
+                .unwrap_or_else(|_| panic!("Failed to read {}", filename));
+            Some(bytecode)
         } else {
             None
         };
         bytecodes.push((preparation_bytecode, bytecode));
     }
+
     let mut total_execution_stats: HashMap<String, u64> = HashMap::default();
     let mut total_execution_time = Duration::from_secs(0);
     let bytecodes_len = bytecodes.len() as u64;
-    //let mut nb_compiled = 0;
+    // let mut nb_compiled = 0;
     for (preparation_bytecode, bytecode) in bytecodes {
-        //let mut rng = rand::thread_rng();
-        //let need_compile = rng.gen_bool(0.5);
+        // let mut rng = rand::thread_rng();
+        // let need_compile = rng.gen_bool(0.5);
         let interface = InterfaceImpl::new_default(
-            Address::from_str("AS12cMW9zRKFDS43Z2W88VCmdQFxmHjAo54XvuVV34UzJeXRLXW9M").unwrap(),
+            Address::from_str(
+                "AS12cMW9zRKFDS43Z2W88VCmdQFxmHjAo54XvuVV34UzJeXRLXW9M",
+            )
+            .unwrap(),
             Some(op_datastore.clone()),
         );
+
         if let Some(preparation_bytecode) = preparation_bytecode {
-            run_main_gc(
+            if let Err(e) = run_main_gc(
                 &interface,
                 RuntimeModule::new(
                     &preparation_bytecode,
-                    u64::MAX,
                     GasCosts::default(),
                     Compiler::CL,
+                    CondomLimits::default(),
                 )
                 .unwrap(),
                 &[],
                 u64::MAX,
                 GasCosts::default(),
-            )
-            .unwrap();
+                CondomLimits::default(),
+            ) {
+                println!("Failed to execute preparation bytecode: {:?}", e);
+                continue;
+            }
         }
-        //let (start, results) = if need_compile {
-        let module =
-            RuntimeModule::new(&bytecode, u64::MAX, GasCosts::default(), Compiler::CL).unwrap();
+
+        let module = RuntimeModule::new(
+            &bytecode,
+            GasCosts::default(),
+            Compiler::CL,
+            CondomLimits::default(),
+        )
+        .unwrap();
         let start = std::time::Instant::now();
-        let results = run_main_gc(&interface, module, &[], u64::MAX, GasCosts::default()).unwrap();
-
-        //println!("Results:");
-        /*println!("");
-        println!("Counters:");
-        for (key, value) in &results.counters {
-            println!("key: {:?}, value: {:?}", key, value);
-        }*/
+
+        let results = match run_main_gc(
+            &interface,
+            module,
+            &[],
+            u64::MAX,
+            GasCosts::default(),
+            CondomLimits::default(),
+        ) {
+            Ok(results) => results,
+            Err(e) => {
+                println!("Failed to execute bytecode: {:?}", e);
```

Comment:
In the final results, do we have the info that a given ABI failed to execute and the gas is not reliable, or do we need to check the prints? I ask because of two effects:
- Knowing that the result of a given ABI is ok
- Knowing that a "bad" result does not influence the rest of the computation (I think launch for exemple is computed based on all ABIs gas costs, so if some of them are wrong, it can propagate)

Reply:
I'm having a loot at this

Reply:
@Leo-Besancon 
After the fix for your other comment and a review of the logs.
Any Abi that faced an issue during their generation or execution is absent from the results.

The original tool was designed to stop at the first error. 
We may: 
1. reintroduce this behavior (it implies we fix around 10 generation issues for AS abi, one execution, and 2 execution issues for wasmv1 (and maybe some missing))
2. check the traces

we could adapt the software but it's quite a bit of work.

I'd go for choice 2 for the moment.